app/assets/javascripts/listings/ListingUnitService.js.coffee

```diff
@@ -29,49 +34,129 @@ ListingUnitService = ($http, ListingConstantsService, ListingIdentityService) ->
     else
       min
 
+  # FIXME: This might be broken by the new format
   Service.combineUnitSummaries = (listing) ->
-    # combined unitSummary is useful e.g. for overall occupancy levels across the whole listing
+    # Combined unitSummary is useful e.g. for overall occupancy levels across the whole listing.
     listing.unitSummaries ?= {}
     combined = _.concat(listing.unitSummaries.reserved, listing.unitSummaries.general)
     combined = _.omitBy(_.uniqBy(combined, 'unitType'), _.isNil)
-    # rename the unitType field to match how individual units are labeled
+    # Rename the unitType field to match how individual units are labeled.
     _.map(combined, (u) -> u.Unit_Type = u.unitType)
     Service._sortGroupedUnits(combined)
 
-  # Given ListingConstantsService.fieldsForUnitGrouping,
-  # combine similar units, and add up the number of available units
   Service._sumSimilarUnits = (units) ->
+    # Given ListingConstantsService.fieldsForUnitGrouping,
+    # combine similar units, and add up the number of available units.
     summaries = []
     # Create an identity function to group by unique price and income
     group = _.groupBy units, (unit) ->
       _.flatten(_.toPairs(_.pick(unit, ListingConstantsService.fieldsForUnitGrouping)))
     _.forEach group, (groupedUnits, id) ->
-      # Summarize each group by combining the unit details + # of units for that type/AMI combo
+      # Summarize each group by combining the unit details + # of units for that type/AMI combo.
       summary = _.pick(groupedUnits[0], ListingConstantsService.fieldsForUnitGrouping)
       summary.total = groupedUnits.length
       summaries.push(summary)
     summaries
 
-  # Group units by AMI % and Unit type
-  # Returns an object of the form:
-  #  {'50 (percent ami)':
-  #     {'1br': [{rent/price, # of units, etc}, ...], '2br': [...]},
-  #   '40 (percent ami)': {...}}
+  Service._getAMIAmount = (incomeList, occupancy) ->
+    if incomeList && incomeList.values?
+      _.find(incomeList['values'], {'numOfHousehold': occupancy})['amount']
+    else
+      console.error('Missing AMIs for occupancy', incomeList, occupancy)
```

Comment:
honestly not sure what our response should be in this case. Without the check it fails with a non-descript error that says that values is undefined with nil.

If we log this as console.error it'll show up in sentry and we'll get the url for the page that the error was raised on too. 

The hope is that that it won't ever be the case that we don't have AMIs that match for the units.